node/src/neighborhood/mod.rs

```diff
@@ -1861,46 +1801,68 @@ impl Neighborhood {
         &mut self,
         message: &UiSetExitLocationRequest,
     ) -> (Vec<ExitLocation>, Vec<String>) {
-        //TODO perform following update of db_countries only in Gossip_Acceptor
         self.user_exit_preferences.db_countries = self.init_db_countries();
-        let mut countries_lack_in_neighborhood = vec![];
+        let mut countries_not_in_neighborhood = vec![];
         (
             message
                 .to_owned()
                 .exit_locations
                 .into_iter()
                 .map(|cc| {
-                    for code in &cc.country_codes {
-                        if self.user_exit_preferences.db_countries.contains(code)
-                            || self.user_exit_preferences.fallback_preference
-                                == FallbackPreference::ExitCountryWithFallback
-                        {
-                            self.user_exit_preferences.exit_countries.push(code.clone());
-                            if self.user_exit_preferences.fallback_preference
-                                == FallbackPreference::ExitCountryWithFallback
-                            {
-                                countries_lack_in_neighborhood.push(code.clone());
-                            }
-                        } else {
-                            countries_lack_in_neighborhood.push(code.clone());
-                        }
-                    }
+                    let requested_country_codes = &cc.country_codes;
+                    countries_not_in_neighborhood
+                        .extend(self.enrich_exit_countries(requested_country_codes));
                     ExitLocation {
                         country_codes: cc.country_codes,
                         priority: cc.priority,
                     }
                 })
                 .collect(),
-            countries_lack_in_neighborhood,
+            countries_not_in_neighborhood,
         )
     }
 
+    fn enrich_exit_countries(&mut self, country_codes: &Vec<String>) -> Vec<String> {
```

Comment:
Slo by v tom nazvu take obsahnout ten druhy efekt, tj. ze to vraci jeste vektor tech zemi, ktere chybi? 

Budto muzes rict "try_add_exit_countries", coz ma podtext, ze ne vzdy to musi vyjit.
Nebo neco na zpusob "enrich_exit_countries_and_collect_missing_ones" nebo podobne.